### DISCUSSION TITLE
How to run testclass with xunit for mobile app automation (appium,c#,xunit).

### DISCUSSION THREAD

**Discussion by pritesh848** [reqnroll/Reqnroll#267] [Ideas]
How to run testclass with xunit, I want to run step defn file class as a testcase but If I add [FACT] attribute to every method in step defn file  the app is launching for every method in step definition file. (Means every step definition method is acting as testcase)
How to run class as a test ? Using xunit

Also plz tell how to generate HTML report ?

**Comment by clrudolphi** [comment]
Hi @pritesh848 welcome to Reqnroll. I suggest you read through the [Quickstart ](https://docs.reqnroll.net/latest/quickstart/index.html) in the Reqnroll documentation.

Your step definition methods will not have Xunit attributes, but instead will need a Reqnroll attribute (Given, When, or Then) that indicates when and how each step definition method is matched with a Scenario step from your Feature file. The class that contains the step definition methods is attributed with a [Binding] attribute.
Reqnroll then generates an XUnit test class (with appropriate XUnit attributes, such as [Fact]) for you. 

Reqnroll does not yet have HTML reporting, but it is on the road-map.
